### DISCUSSION TITLE
Backward chaining

### DISCUSSION THREAD

**Discussion by VitaliyMez** [denkbares/d3web-KnowWE#19] [Q&A]
Hi,

Is it possible to use backward chaining in KnowWE?
Probably there is a possibility to find a sequence of rules fired by the inference engine?

Thank you,
Vitaliy.

**Comment by AlbrechtStriffler** [comment]
Hi Vitaliy,
yes, d3web supports backwards and forward chaining (and a lot more). I'm not sure if that was your question though. Do you want to see which rules have fired for a certain derived fact? Or the facts that contributed to the derivation of another fact? We already show the second one in the tooltips on the object when using the QuickInterview in KnowWE. The tooltip is available in the ShowSolutions- and other markups. If a question or solution has a value, the tooltip will show what other facts contributed to that value.
![image](https://user-images.githubusercontent.com/13681360/201184928-0dc2dc9a-32f5-49fb-8635-ffe5d0340f69.png)

The highlighting in the Rule markup will show which rule actions have fire and which conditions are true/false, based on the current input
![image](https://user-images.githubusercontent.com/13681360/201185324-ddad5787-1fb0-4feb-996d-5890b0387802.png)

Hope this helps... we don't currently have a list of rules that specifically fired for the derivation of a specific fact, but I guess this should also be possible with some work.

Regards
Albrecht
